src/app/(admin)/new_dashboard/guides/hooks/useSearchGuides.ts

```diff
@@ -12,7 +14,16 @@ export const useSearchGuides = (initialGuides: Guide[]) => {
     const matchesTag =
       selectedTag === "All" || guide.categories.includes(selectedTag);
     return matchesQuery && matchesTag;
-  });
+  })
+    .sort((a, b) => {
+      if (sortCriteria === "rating") {
+        const avgRatingA = a.reviews?.reduce((acc, review) => acc + review.rating, 0) ?? 1;
```

Comment:
why copy and paste this function twice and not extract it? 

https://en.wikipedia.org/wiki/Don%27t_repeat_yourself